### PR TITLE
Add State Manager API tests to karma

### LINKED ISSUE #465: Improve VM browser compatibilty / test coverage
With the merge of #461 there is now browser testing done on the VM to improve VM browser compatibility and avoid introducing bugs like #457.

Currently browser tests are not run on all API test files and many files are excluded since tests were not passing, see [karma.conf.js](https://github.com/ethereumjs/ethereumjs-vm/blob/master/karma.conf.js) ``exclude`` directive.

Excluded tests should be integrated and the underlying issues addressed. Initial experiments indicated that there probably will be some deeper investigation needed and it's probably most practical to address step-by-step on separate PRs. It is also likely that these fixes will spawn over different libraries and not only target the VM, so this should be addressed in a holistic way with PRs where necessary.

The following is a list of the test files which need to be activated:

- [ ] ``./tests/api/state/stateManager.js``
- [x] ``./tests/api/state/storageReader.js``
- [x] ``./tests/api/index.js``
- [x] ``./tests/api/runBlock.js'``
- [x] ``./tests/api/runBlockchain.js``

### PR DESCRIPTION
**Pull request by ryanio**
This PR:
1. Adds `tests/api/state/stateManager.js` back to karma.
1. Skips slow test when running in karma: `should generate the genesis state root correctly for mainnet`
1. Modifies test `should generate correct genesis state root for all chains` to `should generate the genesis state root correctly for all other chains` (since mainnet is already tested prior).
1. Runs karma in both headless Firefox and Chrome for increased browser coverage.
1. Removes unneeded `karma-detect-browsers`.

Closes #465

### CODE REVIEW

**Comment by ryanio** [comment]
Sorry about the prettified `tests/api/state/stateManager.js`, the diff is hard to read.

All I changed was:

1. https://github.com/ethereumjs/ethereumjs-vm/pull/653/files#diff-d9351b389fb846285b26d56dcd810b68R161-R165
1. https://github.com/ethereumjs/ethereumjs-vm/pull/653/files#diff-d9351b389fb846285b26d56dcd810b68R185-R186

File: tests/api/state/stateManager.js [hunk 1]
```diff
@@ -55,79 +67,111 @@ tape('StateManager', (t) => {
...
 
-    st.notOk(res)
+      st.end()
+    },
+  )
 
-    st.end()
-  })
+  t.test('should generate the genesis state root correctly for mainnet', async st => {
+    if (typeof window !== 'undefined' && window.__karma__) {
```
Comment (evertonfraga) [inline tests/api/state/stateManager.js]: 👍

File: karma.conf.js [hunk 2]
```diff
@@ -8,15 +8,13 @@ module.exports = function(config) {
...
+    frameworks: ['browserify', 'tap'],
 
     // list of files / patterns to load in the browser
     files: ['./tests/api/**/*.js'],
 
     // list of files / patterns to exclude
-    exclude: [
-      './tests/api/state/stateManager.js', // 4, "# should clear the cache when the state root is set"
-    ],
+    exclude: [],
```
Comment (evertonfraga) [inline karma.conf.js]: 👍

File: tests/api/state/stateManager.js (hunk 1 above)
Comment (ryanio) [reply tests/api/state/stateManager.js]: Good suggestion, I will add the following util:
```js
/**
 * Checks if in a karma test runner.
 * @returns {bool} is running in karma
 */
exports.isRunningInKarma = () => {
  return typeof window !== 'undefined' && window.__karma__
}
```

File: tests/api/state/stateManager.js [hunk 3]
```diff
@@ -55,103 +69,168 @@ tape('StateManager', (t) => {
...
 
-    await generateCanonicalGenesis()
-    let stateRoot = await getStateRoot()
+      st.end()
+    },
+  )
 
-    st.equal(stateRoot.toString('hex'), genesisData.genesis_state_root)
+  t.test('should generate the genesis state root correctly for mainnet', async st => {
+    if (isRunningInKarma()) {
```
Comment (evertonfraga) [inline tests/api/state/stateManager.js]: 👍

**Comment by ryanio** [comment]
Thanks, sounds good. In the future if a file isn't prettified I'll leave it out so the diff is cleaner and run the prettify in a separate PR.

**Pull request merged**